docs/develop/python/interceptors.mdx

```diff
@@ -0,0 +1,193 @@
+---
+id: interceptors
+title: Interceptors - Python SDK
+sidebar_label: Interceptors
+description: Learn how to implement Interceptors in the Temporal Python SDK to manage inbound and outbound SDK calls, enhance tracing, and add authorization to your Workflows and Activities.
+toc_max_heading_level: 2
+keywords:
+  - interceptors
+tags:
+  - Interceptors
+  - Python SDK
+  - Temporal SDKs
+---
+
+The behavior of the Python SDK can be customized in many useful ways by modifying inbound and outbound calls using Interceptors.
+This is similar to the use of "middleware" in web frameworks such as [Django](https://docs.djangoproject.com/en/5.2/topics/http/middleware/), [Starlette](https://www.starlette.io/middleware/), and [Flask](https://flask.palletsprojects.com/en/stable/lifecycle/#middleware).
+
+The methods you implement on your Interceptor classes can perform arbitrary side effects.
+Interceptors can also perform arbitrary modifications to data before it is received by the SDK's "real" implementation, e.g. for context propagation.
```

Comment:
I think "e.g. for context propagation" will go over a lot of readers' heads and so we shouldn't do that so early on in the doc.

So, just a suggestion, but:

```suggestion
Interceptors can also perform arbitrary modifications to incoming and outgoing data before it is received by the SDK's "real" implementation.
```